Move property 'stopClusterMode' from routing config to build config.

t2gran opened the pull request:
Property is needed at build time, and can therefore not be in the routing-config.

The property is stored in the graph object, so it survives serialization and reloading.

See Issue #2558.

abyrd commented:
Hi @t2gran, as I understand it, rather than removing the stop clustering entirely (as previously discussed) you've decided to keep two different modes of stop clustering, but configure them in the proper place in build-config.

t2gran commented:
Yes I fixed it. I think this functionality (cluster by parrentStop) is redundant, but I don´t have the complete picture. I agree that we should try to take things out of OTP, but maybe we should try to do it a bit more controlled - maybe this is something for the steering comity. I suggest we start using deprecation and remove stuff in the next major version. 

The `public final String stopClusterMode` is totally an Enum - I will change it to an Enum. I introduced an Exception when the config value is wrong, the previous code just logged and used the default. Any thoughts on that? I think configuration errors should trigger an error and in the case of an upgrade it must be tested before setting it in production - so it is safe to do so.

abyrd commented:
I agree, it's good to throw exceptions and fail fast when an unexpected value is encountered in the configuration.

t2gran commented:
Since @abyrd and @sdjacobs approved this I toke the liberty and merged it into master, after doing a "just-to-be-sure" test (making sure OTP start and work) locally.

abyrd commented:
Thanks @t2gran, yes when you've got 2 approvals and passing tests it should be fine to merge. This will be even less of a question when we've got separate dev branches.
